pkg/messaging/v1/invoke_method_request.go

```diff
@@ -66,7 +67,17 @@ func (imr *InvokeMethodRequest) WithActor(actorType, actorID string) *InvokeMeth
 
 // WithMetadata sets metadata
 func (imr *InvokeMethodRequest) WithMetadata(md map[string][]string) *InvokeMethodRequest {
-	imr.r.Metadata = GrpcMetadataToInternalMetadata(md)
+	imr.r.Metadata = MetadataToInternalMetadata(md)
+	return imr
+}
+
+// WithFastHTTPHeaders sets fasthttp request headers
+func (imr *InvokeMethodRequest) WithFastHTTPHeaders(header *fasthttp.RequestHeader) *InvokeMethodRequest {
```

Comment:
Please make this a non-pointer.

Reply:
Copying headers in fasthttp is not allowed. 
// Copying Header by value is forbidden. Use pointer to Header instead.

Also I noticed that we are doing through pointer in response code path : https://github.com/dapr/dapr/blob/a63ada63a11c86c51f4aa36c4b081add677af544/pkg/messaging/v1/invoke_method_response.go#L69

Reply:
@yaron2 , @youngbupark , we can not copy headers so need to use pointer way.

Reply:
I find it odd that the `Request` struct doesn't hold the `RequestHeader` as a pointer as it does with many other types.

Reply:
@shalabhms Can we pass requestCtx pointer instead of fasthttp.ResponseHeader?

Reply:
@youngbupark , as you approved now, I am resolving this conversation.  
@yaron2 , can you confirm also ?